flake.nix

```diff
@@ -0,0 +1,448 @@
+                language = "system";
+                pass_filenames = false;
+              };
+            };
+          };
+        };
+
+        packages = {
+          # Building: does a cargo build
+          default = craneLib.cargoBuild (commonArgs // {
+            inherit cargoArtifacts;
+          });
+          MSRV = craneMSRVLib.cargoBuild (commonArgs // MSRVArgs // {
+            cargoArtifacts = cargoArtifactsMSRV;
+          });
+          WASM = craneWASMLib.cargoBuild (commonArgs // WASMArgs // {
+            cargoArtifacts = cargoArtifactsWASM;
+          });
+        };
+        legacyPackages = {
+          ci = {
+            pre-commit-check = checks.pre-commit-check;
+            clippy = checks.clippy;
+            fmt = checks.fmt;
+            audit = checks.audit;
+            latest = {
+              all = checks.latestAll;
+              noDefault = checks.latestNoDefault;
+              noStdBdk = checks.latestNoStdBdk;
+              noStdChain = checks.latestNoStdChain;
+              noStdEsplora = checks.latestNoStdEsplora;
+            };
+            MSRV = {
+              all = checks.MSRVAll;
+              noDefault = checks.MSRVNoDefault;
+              noStdBdk = checks.MSRVNoStdBdk;
+              noStdChain = checks.MSRVNoStdChain;
+              noStdEsplora = checks.MSRVNoStdEsplora;
+            };
+            WASM = {
+              bdk = checks.WASMBdk;
+              esplora = checks.WASMEsplora;
+            };
+            codeCoverage = craneLLVMLib.cargoLlvmCov (commonArgs // {
+              inherit cargoArtifacts;
+              cargoLlvmCovExtraArgs = "--ignore-filename-regex /nix/store --all-features --workspace --lcov --output-path $out -- --test-threads=2";
+            });
+          };
+        };
+
+        devShells = {
+          default = craneLib.devShell {
+            inherit cargoArtifacts;
+            # inherit check build inputs
+            checks = {
+              clippy = checks.clippy;
+              fmt = checks.fmt;
+              default = checks.latest;
+              all = checks.latestAll;
+              noDefault = checks.latestNoDefault;
+              noStdBdk = checks.latestNoStdBdk;
+              noStdChain = checks.latestNoStdChain;
+              noStdEsplora = checks.latestNoStdEsplora;
+            };
+            # dependencies
+            packages = buildInputs ++ [
+              pkgs.bashInteractive
+              pkgs.git
+              pkgs.ripgrep
+              rustTarget
+            ];
+            # pre-commit-checks
+            inherit (self.checks.${system}.pre-commit-check) shellHook;
+
+            # env vars
+            BITCOIND_EXEC = commonArgs.BITCOIND_EXEC;
+            ELECTRS_EXEC = commonArgs.ELECTRS_EXEC;
+          };
+          MSRV = craneMSRVLib.devShell {
+            cargoArtifacts = cargoArtifactsMSRV;
+            # inherit check build inputs
+            checks = {
+              clippy = checks.clippy;
```

Comment:
**nitpick:** This could be `inherit (checks) clippy fmt audit;` which is generally preferable. I think my Nix LSP (`nil`?) flags these and has an LSP-action to fix it, as it's considered a better style.

Reply:
Yes I also use `nil` and there is a code action to do this.
Thanks, refactored.